security/pkg/workload/server.go

```diff
@@ -103,25 +116,29 @@ func (s *SDSServer) StreamSecrets(stream sds.SecretDiscoveryService_StreamSecret
 }
 
 // NewSDSServer creates the SDSServer that registers
-// SecretDiscoveryServiceServer, a gRPC server, which listens on the given
-// Unix Domain Socket.
-func NewSDSServer(udsPath string) (*SDSServer, error) {
+// SecretDiscoveryServiceServer, a gRPC server.
+func NewSDSServer() *SDSServer {
 	var opts []grpc.ServerOption
 	grpcServer := grpc.NewServer(opts...)
 	s := &SDSServer{
-		udsPath:    udsPath,
 		grpcServer: grpcServer,
 	}
 
 	sds.RegisterSecretDiscoveryServiceServer(grpcServer, s)
+	return s
+}
 
+// RegisterUdsPath registers a path for Unix Domain Socket and has
+// SDSServer's gRPC server listen on it.
+func (s *SDSServer) RegisterUdsPath(udsPath string) error {
+	s.udsPaths = append(s.udsPaths, udsPath)
 	_, err := os.Stat(udsPath)
 	if err == nil {
-		return nil, fmt.Errorf("UDS path %v already exists", udsPath)
+		return fmt.Errorf("UDS path %v already exists", udsPath)
```

Comment:
Please use log.Errorf() as above.

Reply:
This is for returning the error string which will be logged by the caller. I thought log.Errorf() is to print it to the error log.